### PR TITLE
Migrated all remaining components to TS

### PR DESCRIPTION
**Pull request by RunDevelopment**

### CODE REVIEW

File: src/common-types.ts [hunk 1]
```diff
@@ -12,14 +12,19 @@ export interface IteratorSize extends Size {
...
+}
 export type Input = {
   type: string;
   label: string;
   optional?: boolean;
-  def?: InputValue;
-  default?: InputValue;
-  options?: { value: InputValue }[];
+  def?: InputSchemaValue;
+  default?: InputSchemaValue;
```
Comment (joeyballentine) [inline src/common-types.ts:26]: This makes me think we should probably standardize whether it's default or def

File: src/helpers/contexts/GlobalNodeState.tsx [hunk 2]
```diff
@@ -576,21 +575,25 @@ export const GlobalProvider = ({
   );
 
   const useInputData = useCallback(
-    (id: string, index: number) => {
+    // eslint-disable-next-line prefer-arrow-functions/prefer-arrow-functions, func-names
+    function <T extends InputValue>(
```
Comment (joeyballentine) [inline src/helpers/contexts/GlobalNodeState.tsx:579]: Just curious why you did it like this

File: src/common-types.ts (hunk 1 above)
Comment (RunDevelopment) [reply src/common-types.ts:26]: Oh, yes.

File: src/helpers/contexts/GlobalNodeState.tsx (hunk 2 above)
Comment (RunDevelopment) [reply src/helpers/contexts/GlobalNodeState.tsx:579]: From the perspective of `useInputData`. the input data is a `InputValue` (string or number (at least for now)), but from the perspective of e.g. `FileInput`, the input data has to be a string. Using generics, we resolve this perspective problem.

Comment (RunDevelopment) [reply src/helpers/contexts/GlobalNodeState.tsx:579]: Ah, about that. As it turns, TSX will interpret `<T>() => {}` as the HTML tag `<T>` followed by an arrow function. That's why I had to use `function`.

Comment (joeyballentine) [reply src/helpers/contexts/GlobalNodeState.tsx:579]: Makes sense

**Pull request merged**